File: hangman/hangman.go
package main

import "fmt"

// import (
// 	"bufio"
// 	"fmt"
// 	"math/rand"
// 	"os"
// 	"strings"
// 	"time"
// )

// func main() {
// 	wordList := [10]string{
// 		"chair",
// 		"table",
// 		"mouse",
// 		"keyboard",
// 	}

// 	rand.Seed(time.Now().UnixNano())
// 	word := wordList[rand.Intn(10)]

// 	blankList := make([]string, len(word))

// 	for i := 0; i < len(blankList); i++ {
// 		blankList[i] = "_"
// 	}

// 	fmt.Println(blankList)
// 	chances := len(word) + 3
// 	for {
// 		chances--
// 		reader := bufio.NewReader(os.Stdin)
// 		fmt.Println("Guess any letter!")
// 		guess, err := reader.ReadString('\n')
// 		guess = guess[:1]
// 		if err != nil {
// 			fmt.Println("An error occured while reading input. Please try again", err)
// 			continue
// 		}
// 		// logic goes after this, to add  guess if it exists in word to blanklist
// 		slicedWord := word
// 		lenSlicedOut := 0
// 		for i := 0; i < strings.Count(slicedWord, guess); i++ {
// 			blankList[strings.IndexAny(slicedWord, guess)+lenSlicedOut] = guess
// 			slicedWord = word[strings.IndexAny(slicedWord, guess):]
// 			lenSlicedOut = len(word) - len(slicedWord)
// 			fmt.Println("Sliced out: ", lenSlicedOut)
// 			fmt.Println("Sliced word: ", slicedWord)

// 		}

// 		if strings.Join(blankList, "") == word {
// 			break
// 		}

// 		fmt.Println("Your guess is", guess)
// 		fmt.Println(blankList)
// 		if chances == 0 {
// 			break
// 		}

// 	}
// 	arrToString := strings.Join(blankList, "")
// 	fmt.Println(arrToString)
// 	if arrToString == word {
// 		fmt.Println("Great Work! You guessed the word correctly.")
// 	} else {
// 		fmt.Println("Oops, You ran out of chances! The word was actually ", word, ".")
// 	}

// }

func foo() {
	fmt.Println("H")
}
func main() {

}

File: hangman/hangman.py
# hangman.py
#hangman game implementation in python

import random

#create word list
wordList = [
    ("keyboard", "Used to Type")  , 
    ("mouse","Used to scroll"),
    ("laptop","Used to code"),
    ("table","a piece of furniture with a flat top used for writing"),
    ("chair","Something you sit on"),
    ("headphones","Used to listen to music"),
]

chances = 1

 
exit = "y"
while exit != "n":
    while chances > 0 :
        #Randomly pick one word  
        randomChoice = random.randrange(len(wordList))
        word = wordList[randomChoice][0]

        #Set chances to len+1
        chances = len(word) +1
        blankList = ["_"] *len(word)
        print()
        print()
        print("HINT: ", wordList[randomChoice][1])
        for i in range(chances):
            for i in blankList:
                print(i, end=" ")
            print()
            print()
            print("There are ", chances, "chances left.")
            print()
            guess = input("Please enter a letter to guess! \n")
            print()
            numOccurence = word.count(guess)
            wordSliced = word
            lengthSlicedOut = 0
            for i in range(numOccurence):
                blankList[wordSliced.find(guess) + lengthSlicedOut] = guess
                lengthSlicedOut += len(word[:numOccurence+1])
                if numOccurence >1:
                    wordSliced = word[numOccurence+1:]
            if blankList == list(word):
                print("Correct! The word was ",word )
                break
            
            chances = chances-1  

    if blankList != list(word):
        print("Oops! The word was ",word )
        print("Try again!")
    exit = input("Do you wish to play again?(y/n)")
    chances = 1

                

